.github/workflows/deploy.yml

```diff
@@ -0,0 +1,40 @@
+name: Deploy to VSCode marketplace
+
+on:
+  release:
+    types: [published]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  deploy:
+    runs-on: windows-latest
+    timeout-minutes: 5
+    
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2.1.0
+      with:
+        submodules: true
+    
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v1.4.2
+      with:
+        node-version: '12.x'
+    
+    - name: Check release version matches package.json
+      uses: ./.github/actions/check-package-version
+
+    - name: Install dependencies
+      run: npm install
+
+    - name: Publish to VSCode Marketplace
+      env:
+        PAT: ${{ secrets.MARKETPLACE_PAT }}
+      run: npx vsce publish -p "$PAT"
+
+    - name: Delete tag on failure
+      if: failure()
+      run: |
+        git push origin :${{ github.ref }}
```

Comment:
This pushes the build to the tag?

Reply:
It pushes an empty ref to the remote, thus deleting the remote tag so we can try again. https://stackoverflow.com/questions/5480258/how-to-delete-a-remote-tag/5480292#5480292

Reply:
ooh, that's new to me!